pkg/upgrade/handle_batch.go

```diff
@@ -41,10 +53,34 @@ func (ctl *Controller) handleJobs(ctx context.Context) error {
 						}
 					}
 				}
+				if upgradejob.ConditionComplete.IsTrue(obj) {
+					return obj, enqueueOrDelete(jobs, obj, upgradejob.ConditionComplete)
+				}
+				if upgradejob.ConditionFailed.IsTrue(obj) {
+					return obj, enqueueOrDelete(jobs, obj, upgradejob.ConditionFailed)
+				}
```

Comment:
I did purposely leave these under the "if the job has a plan name label" flow control because right now they are the only kind of job supported by this controller but that may not always be the case.